### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Ukaypromise**
#### Project requirements
For this PR the following requirements were adhered to:

- [x] Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.

- [x]  Follow SOLID principles.

- [x] Pay special attention to the "Single-responsiblity Principle".

- [x] Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).

- [x] Extract some pieces of logic into separate class(es).

### CODE REVIEW

File: src/rental_operations.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    end
+    book_index = gets.chomp.to_i
+
+    puts
+    puts 'Select a person from the following list by number (not id)'
+    puts '========================================================================'
+    @people.each_with_index do |person, index|
+      puts "#{index}) #{person} Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    puts '========================================================================'
```
Comment (1GORDON) [inline src/rental_operations.rb]: Great job team 🤝

- [ ] According to the Single Responsibility Rule, `all requests for input and input validation should've been placed in the main.rb file`. Kindly check out this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more guidance.

File: src/person_operations.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    if person_type == '1'
+      print 'Has parent permission? [Y/N]: '
+      parent_permission = gets.chomp
+      parent_permission = parent_permission.downcase == 'y'
+      student = Student.new(name, parent_permission, age)
```
Comment (1GORDON) [inline src/person_operations.rb]: Great job team 🤝

- [ ] According to the Single Responsibility Rule, `all requests for input and input validation should've been placed in the main.rb file`. Kindly check out this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more guidance.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @Ukaypromise & @Aseadd ✋ 

Good job so far!

## Highlights!
- Followed Gitflow ✔️
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️

- [ ] According to the Single Responsibility Rule. kindly make sure all requests for input and input validation should've been placed in the main.rb file. You may implement this across all files.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/person_operations.rb (hunk 2 above)
Comment (Aseadd) [reply src/person_operations.rb]: Input and validation request placed on the main.rb file

File: src/rental_operations.rb (hunk 1 above)
Comment (Aseadd) [reply src/rental_operations.rb]: Change made accordingly

**Review by MahmoudBakr23 — APPROVED** [review]
Hi team😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done on making the required changes from the previous reviewer.
- Great communication

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**